### PR TITLE
Enable RUF013 (implicit optional)

### PR DESCRIPTION
**Pull request by alexrudd2**
The recent PRs eliminating implicit optional are now enough to satisfy `ruff` (although not yet `mypy`)

### CODE REVIEW

**Review by janiversen — CHANGES_REQUESTED** [review]
Seems this PR have a problem, we currently do not have on dev.

**Comment by alexrudd2** [comment]
> Seems this PR have a problem, we currently do not have on dev.

Very strange, since there are no code changes.  I think `dev` is broken, but the cache has been hiding it.  

I manually deleted the Linux/3.8 cache and reran the workflow manually, and [it fails](https://github.com/pymodbus-dev/pymodbus/actions/runs/6805365296/job/18505192239).  The tests also fail on my Mac.  I will bisect.

**Comment by alexrudd2** [comment]
Triaged as https://github.com/pymodbus-dev/pymodbus/issues/1883

**Comment by janiversen** [comment]
I just tested dev without cache, no problem on my computer.

**Comment by alexrudd2** [comment]
> I just tested dev without cache, no problem on my computer.

(I think you realize this already, but your computer would have broken upon upgrading to `aiohttp==3.9.0b1`)

Anyways, that is solved.   I can't merge until you re-review with approval.

**Review by janiversen — APPROVED** [review]
Thanks

**Pull request merged**
